pkg/deploy/upgrade.go

```diff
@@ -47,6 +52,27 @@ func (d *deployer) waitForRPReadiness(ctx context.Context, vmssName string) erro
 	}, ctx.Done())
 }
 
+// removeOldAlertRules removes alert rules without the location in the name
+func (d *deployer) removeOldAlertRules(ctx context.Context) error {
+	d.log.Print("removing old alerts")
+	alertRules, err := d.alertrules.ListByResourceGroup(ctx, d.config.ResourceGroupName)
+	if err != nil {
+		return err
+	}
+
+	for _, alertRule := range *alertRules.Value {
+		switch *alertRule.Name {
+		case "rp-availability-alert", "rp-degraded-alert", "rp-vnet-alert":
+			_, err = d.alertrules.Delete(ctx, d.config.ResourceGroupName, *alertRule.Name)
```

Comment:
What is the behaviour of this API while deleting something which does not exist? Does it give error `NotFound` or noop?

Reply:
It responds with 204 (tested with az-cli + https://docs.microsoft.com/en-us/rest/api/monitor/metricalerts/delete )